hydromt/data_adapter/dataset.py

```diff
@@ -0,0 +1,431 @@
+        data_name : str
+            Name of output file without extension.
+        variables : list of str, optional
+            Names of Dataset variables to return. By default all dataset variables
+            are returned.
+        time_tuple : tuple of str, datetime, optional
+            Start and end date of period of interest. By default the entire time period
+            of the dataset is returned.
+        driver : str, optional
+            Driver to write file, e.g.: 'netcdf', 'zarr', by default None
+        **kwargs
+            Additional keyword arguments that are passed to the `to_zarr`
+            function.
+
+        Returns
+        -------
+        fn_out: str
+            Absolute path to output file
+        driver: str
+            Name of driver to read data with, see
+            :py:func:`~hydromt.data_catalog.DataCatalog.get_dataset`
+        """
+        obj = self.get_data(
+            time_tuple=time_tuple,
+            variables=variables,
+            logger=logger,
+            single_var_as_array=variables is None,
+        )
+        if driver is None or driver == "netcdf":
+            fn_out = netcdf_writer(
+                obj=obj, data_root=data_root, data_name=data_name, variables=variables
+            )
+        elif driver == "zarr":
+            fn_out = zarr_writer(
+                obj=obj, data_root=data_root, data_name=data_name, **kwargs
+            )
+        else:
+            raise ValueError(f"Dataset: Driver {driver} unknown.")
+
+        return fn_out, driver
+
+    def get_data(
+        self,
+        variables: Optional[List[str]] = None,
+        time_tuple: Optional[Tuple[str | datetime]] = None,
+        single_var_as_array: Optional[bool] = True,
+        logger: Optional[logging.Logger] = logger,
+    ) -> xr.Dataset:
+        """Return a clipped, sliced and unified Dataset.
+
+        For a detailed description see:
+        :py:func:`~hydromt.data_catalog.DataCatalog.get_dataset`
+        """
+        # load data
+        fns = self._resolve_paths(variables, time_tuple)
+        ds = self._read_data(fns, logger=logger)
+        self.mark_as_used()  # mark used
+        # rename variables and parse data and attrs
+        ds = self._rename_vars(ds)
+        ds = self._set_nodata(ds)
+        ds = self._shift_time(ds, logger=logger)
+        # slice
+        ds = DatasetAdapter._slice_data(ds, variables, time_tuple, logger=logger)
+        # uniformize
+        ds = self._apply_unit_conversion(ds, logger=logger)
+        ds = self._set_metadata(ds)
+        # return array if single var and single_var_as_array
+        return self._single_var_as_array(ds, single_var_as_array, variables)
+
+    def _read_data(self, fns, logger=logger):
+        kwargs = self.driver_kwargs.copy()
+        if len(fns) > 1 and self.driver in ["zarr"]:
+            raise ValueError(
+                f"Dataset: Reading multiple {self.driver} files is not supported."
+            )
```

Comment:
we should probably document, or implement this if it is actually possible for the ther formats. Could you make tickets as necessary?

Reply:
See #651